content/master/concepts/compositions.md

```diff
@@ -29,385 +65,210 @@ Crossplane has four core components that users commonly mix up:
-{{< /hint >}}
-  
-
-For example, a Composition can define a template to create a virtual machine 
-and an associated storage bucket at the same time. 
-
-The {{<hover label="resources" line="4">}}resources{{</hover>}} field lists the 
-individual resources with a 
-{{<hover label="resources" line="5">}}name{{</hover>}}.  
-This name identifies the
-resource inside the Composition and isn't related to the external name used with
-the Provider.
-
-#### Template a managed resource
-The contents of the 
-{{<hover label="resources" line="6">}}base{{</hover>}} are identical to
-creating a standalone [managed resource]({{<ref "./managed-resources">}}).
-
-This example uses 
-[Upbound's Provider AWS](https://marketplace.upbound.io/providers/upbound/provider-aws/v0.35.0)
-to define a 
-S3 storage {{<hover label="resources" line="8">}}Bucket{{</hover>}} and 
-EC2 compute {{<hover label="resources" line="15">}}Instance{{</hover>}}.
-
-After defining the 
-{{<hover label="resources" line="7">}}apiVersion{{</hover>}} and 
-{{<hover label="resources" line="8">}}kind{{</hover>}}, define the 
-{{<hover label="resources" line="10">}}spec.forProvider{{</hover>}} fields
-defining the settings of the resource.
-
-```yaml {copy-lines="none",label="resources"}
-apiVersion: apiextensions.crossplane.io/v1
-kind: Composition
-spec:
-  resources:
-    - name: StorageBucket
-      base:
-        apiVersion: s3.aws.upbound.io/v1beta1
-        kind: Bucket
-        spec:
-          forProvider:
-            region: "us-east-2"
-    - name: VM
-      base:
-        apiVersion: ec2.aws.upbound.io/v1beta1
-        kind: Instance
-        spec:
-          forProvider:
-            ami: ami-0d9858aa3c6322f73
-            instanceType: t2.micro
-            region: "us-east-2"
-```
-
-When a [Composite Resource]({{<ref "./composite-resources" >}}) uses this
-Composition template, the Composite Resource creates two new managed resources
-with all the provided 
-{{<hover label="resources" line="17">}}spec.forProvider{{</hover>}} settings.
+Composition functions (or just functions, for short) are Crossplane extensions
+that template Crossplane resources. Crossplane calls the composition functions
+to determine what resources it should create when you create a composite
+resource (XR).
 
-The {{<hover label="resources" line="16">}}spec{{</hover>}}
-supports any settings used in a managed resource including
-applying `annotations` and `labels` or using a specific 
-`providerConfigRef`.
-
-{{<hint "note" >}}
-Compositions allow applying a `metadata.name` to a resource's 
-{{<hover label="resources" line="16">}}spec{{</hover>}} but ignores it. The
-`metadata.name` field doesn't influence the name
-of the managed resource in Crossplane or the external resource inside the 
-Provider.
-
-Use the `crossplane.io/external-name` annotation on the resource to influence
-the external resource name.
+{{<hint "tip" >}}
+The Crossplane community has built lots of functions that let you template
```

Comment:
Should we call out https://github.com/crossplane-contrib/function-kcl as well considering its popularity.

Reply:
Done.